integration/aggregate_documents_compat_test.go

```diff
@@ -1562,3 +1562,47 @@ func TestAggregateCompatProject(t *testing.T) {
 
 	testAggregateStagesCompat(t, testCases)
 }
+
+func TestAggregateCompatUnset(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]aggregateStagesCompatTestCase{
+		"InvalidType": {
+			pipeline: bson.A{
+				bson.D{{"$unset", "invalid"}},
```

Comment:
Why is is called `InvalidType`? This example looks like a valid `$unset` to me, where we unset a field called `invalid`. If it's about an invalid type, then maybe we can use something like  `bson.D{{"$unset",  false}},` (then the type `bool` is really invalid. 

Let's also add one more test where wrong types are set in an array. For example, something like `{$unset: ["v", int32(42), false]}`